cn_indian_payroll/cn_indian_payroll/overrides/payroll_configuration.py

```
@@ -1,119 +1,297 @@
+    component.custom_regime = data.get("regime")
+    
+    component.formula = data.get("formula")
+    component.condition = data.get("condition")
+    component.custom_sequence = data.get("sequence")
+    component.component_type = name
+    component.insert()
 
-                custom_field_doc.save()
-                frappe.db.commit()  
+@frappe.whitelist()
+def get_salary_component(data=None, component=None, custom_field=None):
```

Ruff (0.8.2)

`get_salary_component` is too complex (15 > 10)

(C901)